main.rb

```diff
@@ -0,0 +1,178 @@
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      new_rental
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_selected = gets.chomp.strip.to_i
+    case person_selected
+    when 1
+      create_student
+      @id += 1
+    when 2
+      new_teacher
+      @id += 1
+    else
+      'Invalid Selection. Returning to main menu'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    if age <= 0
+      @id -= 1
+      return 'Wrong input for age. Returning to main menu'
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Has Parent Permission? [Y/N]: '
+    permission = gets.chomp.strip.upcase
+    case permission
+    when 'Y', 'YES'
+      permission = true
+    when 'N', 'NO'
+      permission = false
+    end
+    student = Student.new(@id, age, nil, name, parent_permission: permission)
+    @people << student
+    puts 'Person created successfully'
+  end
+
+  def new_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    if age <= 18
+      @id -= 1
+      return 'Wrong input for age all teachers should be at least 18. Returning to main menu'
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp.strip.capitalize
+    p "test teacher age: #{age}"
+    teacher = Teacher.new(@id, age, name, specialization)
+    @people << teacher
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp.strip.capitalize
+    print 'Author: '
+    author = gets.chomp.strip.capitalize
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def list_all_people
+    @people.each_with_index do |person, index|
+      p "test person: #{person.class}"
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_all_books
+    @books.each_with_index { |book, index| puts %{#{index}) Title: "#{book.title}", Author: #{book.author}} }
+  end
+
+  def new_rental
+    puts 'Select a book from the following list by number'
+    list_all_books
+    book_selected = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    list_all_people
+    person_selected = gets.chomp.to_i
+    puts 'Date (YYYY/MM/DD) : '
+    date = gets.chomp.strip
+    p 'test rental'
+    Rental.new(date, @people[person_selected], @books[book_selected])
+    puts 'Rental created successfully'
+  end
+
+  def list_person_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals: '
+    @people.each do |person|
+      next unless person.id == id
+
+      person.rentals.each do |rental|
+        puts %(Date: #{rental.date}, Book "#{rental.book.title}" by #{rental.book.author})
+      end
+    end
+  end
+end
```

Comment:
- [ ] **Required:** This method breaks your code. When I attempt to list all rentals, I get this error below. Kindly fix it.

![image](https://user-images.githubusercontent.com/27709832/200153929-d1ee8e5b-3c40-4e2b-82a7-7eb57ff187a0.png)